sklearn/datasets/tests/test_openml.py

```diff
@@ -0,0 +1,413 @@
+"""Test the openml loader.
+"""
+import gzip
+import json
+import numpy as np
+import os
+import scipy.sparse
+import sklearn
+
+from sklearn.datasets import fetch_openml
+from sklearn.datasets.openml import _get_data_features
+from sklearn.utils.testing import (assert_warns_message,
+                                   assert_raise_message)
+from sklearn.externals.six import string_types
+from sklearn.externals.six.moves.urllib.error import HTTPError
+
+
+currdir = os.path.dirname(os.path.abspath(__file__))
+
+
+def fetch_dataset_from_openml(data_id, data_name, data_version,
+                              target_column_name,
+                              expected_observations, expected_features,
+                              exptected_data_dtype, exptected_target_dtype,
+                              expect_sparse, compare_default_target):
+    data_by_name_id = fetch_openml(name=data_name, version=data_version,
+                                   cache=False)
+    assert int(data_by_name_id.details['id']) == data_id
+
+    fetch_openml(name=data_name, cache=False)
+    # without specifying the version, there is no guarantee that the data id
+    # will be the same
+
+    # fetch with dataset id
+    data_by_id = fetch_openml(data_id=data_id, cache=False,
+                              target_column_name=target_column_name)
+    assert data_by_id.details['name'] == data_name
+    assert data_by_id.data.shape == (expected_observations, expected_features)
+    if isinstance(target_column_name, str):
+        # single target, so target is vector
+        assert data_by_id.target.shape == (expected_observations, )
+    elif isinstance(target_column_name, list):
+        # multi target, so target is array
+        assert data_by_id.target.shape == (expected_observations,
+                                           len(target_column_name))
+    assert data_by_id.data.dtype == exptected_data_dtype
+    assert data_by_id.target.dtype == exptected_target_dtype
+
+    if compare_default_target:
+        # check whether the data by id and data by id target are equal
+        data_by_id_default = fetch_openml(data_id=data_id, cache=False)
+        if data_by_id.data.dtype == np.float64:
+            np.testing.assert_almost_equal(data_by_id.data,
```

Comment:
Use `assert_allclose` instead (`assert_almost_equal` uses digits precision which is not good if the data is not normalized)